src/commands/dependencies/__tests__/upload.test.ts

```diff
@@ -0,0 +1,405 @@
+
+    expect(stderr[0]).toEqual(
+      chalk.red(`Missing ${chalk.bold('--source')} parameter. Supported values are: ${chalk.bold('snyk')}`)
+    )
+    expect(stdout).toEqual([''])
+
+    expect(code).toBe(1)
+  })
+
+  test('exits if invalid --source parameter', async () => {
+    const {context, code} = await runUploadDependenciesCommand(
+      './src/commands/dependencies/__tests__/fixtures/dependencies',
+      {
+        apiKey: 'DD_API_KEY_EXAMPLE',
+        appKey: 'DD_APP_KEY_EXAMPLE',
+        dryRun: true,
+        releaseVersion: '1.234',
+        service: 'my-service',
+        source: 'unknown-source',
+      }
+    )
+    const stdout = context.getStdoutBuffer().split(os.EOL)
+    const stderr = context.getStderrBuffer().split(os.EOL)
+
+    expect(stderr[0]).toEqual(
+      chalk.red(`Unsupported ${chalk.bold('--source')} unknown-source. Supported values are: ${chalk.bold('snyk')}`)
+    )
+    expect(stdout).toEqual([''])
+
+    expect(code).toBe(1)
+  })
+
+  test("exits if file doesn't exist", async () => {
+    const filePath = './src/commands/dependencies/__tests__/fixtures/unknown-dependencies'
+    const resolvedFilePath = path.resolve(filePath)
+    const {context, code} = await runUploadDependenciesCommand(filePath, {
+      apiKey: 'DD_API_KEY_EXAMPLE',
+      appKey: 'DD_APP_KEY_EXAMPLE',
+      dryRun: true,
+      releaseVersion: '1.234',
+      service: 'my-service',
+      source: 'snyk',
+    })
+    const stdout = context.getStdoutBuffer().split(os.EOL)
+    const stderr = context.getStderrBuffer().split(os.EOL)
+
+    expect(stderr[0]).toEqual(chalk.red(`Cannot find "${resolvedFilePath}" file.`))
+    expect(stdout).toEqual([''])
+
+    expect(code).toBe(2)
+  })
+
+  test('shows warning if missing --release-version parameter', async () => {
+    const {context, code} = await runUploadDependenciesCommand(
+      './src/commands/dependencies/__tests__/fixtures/dependencies',
+      {
+        apiKey: 'DD_API_KEY_EXAMPLE',
+        appKey: 'DD_APP_KEY_EXAMPLE',
+        dryRun: true,
+        service: 'my-service',
+        source: 'snyk',
+      }
+    )
+    const stdout = context.getStdoutBuffer().split(os.EOL)
+    const stderr = context.getStderrBuffer().split(os.EOL)
+
+    expect(stderr).toEqual([''])
+    const releaseVersion = chalk.bold('--release-version')
+    expect(stdout[0]).toEqual(
+      chalk.yellow('┌──────────────────────────────────────────────────────────────────────────────────────┐')
+    )
+    expect(stdout[1]).toEqual(
+      chalk.yellow(`│ Missing optional ${releaseVersion} parameter.                                        │`)
+    )
+    expect(stdout[2]).toEqual(
+      chalk.yellow('│ The analysis may use out of date dependencies and produce false positives/negatives. │')
+    )
+    expect(stdout[3]).toEqual(
+      chalk.yellow('└──────────────────────────────────────────────────────────────────────────────────────┘')
+    )
```

Comment:
Is this necessary, this seems hard to maintain.
Once you update the layout you'll have to update the test too.
Could we only check for the important part of the output we're looking for?

Something like:
```suggestion
    const output = stdout.join('');
    expect(output.includes('The analysis may use out of date dependencies and produce false positives/negatives.')).toBe(true);
    expect(output.includes(`Missing optional ${releaseVersion} parameter.`)).toBe(true);
```

Or is the layout something we're testing too?